awslabs/aws-lambda-powertools-typescript
# docs: update sam example to nodejs18

## Description of your changes

As detailed in #1195, the SAM Examples in the repo needed some changes to start using Node.js 18.

This PR builds upon the changes #1180 and:
- Sets functions runtime to `nodejs18x`
- Updates bundling configs to exclude `@aws-sdk/*` packages (they're now part of the runtime)
- Moves functions code back to the folder (symlinks were not working well with `jest`)
- Light updates to function's code & dependencies
- Light updates to README file

### How to verify this change

See unit tests in the PR that prove that the examples build successfully.

I have also manually deployed & tested the sample in an AWS account and verified that they work as intended.

### Related issues, RFCs



**Issue number:** #1195

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist


### Breaking change checklist


---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
